### DISCUSSION TITLE
LINQ all in query for property of collection entities

### DISCUSSION THREAD

**Discussion by danielwinkler** [ravendb/ravendb#17475] [Q&A]
Dear all,

For an entity of `MyEntity`

```csharp
class Tag
{
    public string Name { get; set; }
    public string Language { get; set; }
}

class MyEntity
{
    public string Id { get; set; }
    public List<Tag> Tags { get; set; }
}
```

I can filter with RQL

```SQL
from "MyEntities" 
where Tags[].Language all in ("en-US","de-DE")
```

what is the corresponding CSharp code for that? I tried:

```csharp
var query = from myEntity in DbSession.Query<MyEntity>()
            where myEntity.Tags.Select(x => x.Language).ContainsAll(new[] { "en-US", "de-DE" })
            select myEntity;
```

but the `Tags.Select(x => x.Language)` throws an exception that is cannot be translated.

As a workaround i can use `Advanced.DocumentQuery` with `ContainsAll`, but I was wondering if it works with a normal query too.

Thanks for any help in advance

**Comment by ayende** [comment]
```
var query = from myEntity in DbSession.Query<MyEntity>()
            where myEntity.Tags.Any(x => x.Language,In(new[] { "en-US", "de-DE" }))
            select myEntity;
```

**Reply by danielwinkler** [reply]
Thank you @ayende ,

I forgot to to add the `all` in the list RQL, but I would like that each of them matches any of the languages (the question is updated now).

While the `Any` works as you laid out, when i try `All` I get an error.
```
var query = from myEntity in DbSession.Query<MyEntity>()
            where myEntity.Tags.All(x => x.Language.In(new[] { "en-US", "de-DE" }))
            select myEntity;
```

Is there a way for the `ContainsAll` / `all in` too?

**Reply by garayx** [reply]
what error do you get?

**Reply by ayende** [reply]
`ContainsAll` should do it, see:

https://github.com/ravendb/ravendb/blob/3b68c045320cd9405d59e0eabbcc2f6bb19d518b/test/SlowTests/Tests/Linq/Contains.cs#L98

**Reply by danielwinkler** [reply]
Thanks Oren, but i do not have a string[] at hand, I only have a an array of objects with a string property. (`Tags[].Language`)

My current approach is
```
        query
            .ToAsyncDocumentQuery()
            .ContainsAll(propertyName + "[]." + childName, values)
            .ToQueryable();
```

**Comment by danielwinkler** [comment]
> what error do you get?

`System.NotSupportedException: 'Could not understand expression: from 'MyEntities' where `

```
Inner Exception
NotSupportedException: Method not supported: All
```

**Reply by ayende** [reply]
You may need to create a static index for this, rather than rely on the automatic indexes.
